### PR TITLE
Several countdown fixes

### PR DESCRIPTION
**Pull request by jluhrs**
Countdowns are not so pretty when dealing with the real world.

1. Seqexec now uses the FLAMINGOS-2 provided countdown.
2. Guarded against limit cases, like the countdown not having a valid value for several seconds after starting the observation.
3. I had to implement a special case for FLAMINGOS-2, because its countdown does not count down the exposition, but the whole observe command.

### CODE REVIEW

File: modules/seqexec/server/src/main/scala/seqexec/server/EpicsUtil.scala [hunk 1]
```diff
@@ -245,4 +246,15 @@ object EpicsUtil {
...
+                              obsState: F[Option[CarStateGeneric]]): Stream[F, Progress] =
+    ProgressUtil.fromFOption(_ => (remT, obsState).mapN { case (rem, st) =>
+      for{
+        c <- rem
+        s <- st
+        if s.isBusy
+      } yield Progress(if(total>c) total else c, RemainingTime(c))
+    }).dropWhile(_.remaining.self.value === 0.0) // drop leading zeros
+      .takeThrough(_.remaining.self.value > 0.0) // drop all tailing zeros but the first one
+
```
Comment (jluhrs) [inline modules/seqexec/server/src/main/scala/seqexec/server/EpicsUtil.scala]: It is only active while the exposure is in progress, removes leading and tailing zeros, and never produces a `Progress`  where the remaining time is higher than the total.

File: modules/seqexec/server/src/main/scala/seqexec/server/flamingos2/Flamingos2ControllerEpics.scala [hunk 2]
```diff
@@ -41,77 +39,65 @@ object Flamingos2ControllerEpics extends Flamingos2Controller {
...
+    case LyotWheel.H2         => "Hart2_G5834"
+  }
+
+  implicit val encodeGrismPosition: EncodeEpicsValue[Grism, String] = EncodeEpicsValue {
+    case Grism.Open    => "Open"
+    case Grism.R1200HK => "HK_G5802"
+    case Grism.R1200JH => "JH_G5801"
+    case Grism.R3000   => "R3K_G5803"
+    case Grism.Dark    => "DK_G5804"
+  }
```
Comment (jluhrs) [inline modules/seqexec/server/src/main/scala/seqexec/server/flamingos2/Flamingos2ControllerEpics.scala]: Idea gracefully pointed that those were easier to write like this.

File: modules/seqexec/server/src/main/scala/seqexec/server/flamingos2/Flamingos2ControllerEpics.scala [hunk 3]
```diff
@@ -148,8 +134,22 @@ object Flamingos2ControllerEpics extends Flamingos2Controller {
...
+          if obst.isBusy
+          rem <- Flamingos2Epics.instance.countdown.map(_.seconds)
+        } yield Progress(m, RemainingTime(rem))
+        (m, p)
+      }
+    })
+    s(total).dropWhile(_.remaining.self.value === 0.0) // drop leading zeros
+      .takeThrough(_.remaining.self.value > 0.0) // drop all tailing zeros but the first one
+
+  }
```
Comment (jluhrs) [inline modules/seqexec/server/src/main/scala/seqexec/server/flamingos2/Flamingos2ControllerEpics.scala]: FLAMINGOS-2 starts with a countdown value higher than the exposure time. I used a `State` to carry that value as the new `total`.

File: modules/seqexec/server/src/main/scala/seqexec/server/EpicsUtil.scala (hunk 1 above)
Comment (cquiroz) [reply modules/seqexec/server/src/main/scala/seqexec/server/EpicsUtil.scala]: Neat

File: modules/seqexec/server/src/main/scala/seqexec/server/flamingos2/Flamingos2ControllerEpics.scala (hunk 2 above)
Comment (cquiroz) [reply modules/seqexec/server/src/main/scala/seqexec/server/flamingos2/Flamingos2ControllerEpics.scala]: Should we consider making these enums from the db like on gem?

File: modules/seqexec/server/src/main/scala/seqexec/server/gmos/GmosEpics.scala [hunk 4]
```diff
@@ -194,13 +192,9 @@ class GmosEpics(epicsService: CaService, tops: Map[String, String]) {
...
 
-  private val observeCAttr: Option[CaAttribute[CarState]] = Option(dcState.addEnum("observeC",
-    GMOS_TOP + "dc:observeC", classOf[CarState]))
-  def observeC: Option[CarState] = observeCAttr.map(_.value)
-
-  def waitObserve(t: Time): SeqAction[CarState] = observeCAttr.map(attr =>
-    waitForValues(attr, List(CarState.IDLE, CarState.ERROR, CarState.PAUSED),t , "GMOS observe state")).getOrElse(
-    SeqAction.fail(SeqexecFailure.Unexpected(s"Cannot access ${GMOS_TOP}dc:observeC")))
+  private val observeCAttr: CaAttribute[CarState] = dcState.addEnum("observeC",
+    GMOS_TOP + "dc:observeC", classOf[CarState])
```
Comment (cquiroz) [inline modules/seqexec/server/src/main/scala/seqexec/server/gmos/GmosEpics.scala]: `s"$GMOS_TOP...`?

**Pull request merged**